geonode/commands/tests/test_views.py

```diff
@@ -0,0 +1,53 @@
+#########################################################################
+#
+# Copyright (C) 2021 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+
+from rest_framework.test import APITestCase
+from rest_framework import status
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+
+
+class CommandViewTests(APITestCase):
+    """
+    - Test the Permissions vs Response Status
+    - Test the Validations done inside the POST
+    """
+    def setUp(self):
+        self.url = "geonode.commands:enque"
+        self.standard_user = get_user_model().objects.create(
+            username="test", email="test@test.com"
+        )
+        self.admin_user, _ = get_user_model().objects.get_or_create(
+            username="super", is_staff=True, is_superuser=True
+        )
+
+
+    def test_permissions(self):
+        cases = (
```

Comment:
I'll avoid adding more tests cases in a single test.
Potentially here you have 3 different tests. Merging them all together will be a mess if you have to debug, plus is easier to read them even with the verbose. Potentially here you have:

```python
def test_anonymous_users_is_unauthorized(self):
    '''
    Anonymous users cannot access to the API
    '''
    pass
def test_normal_user_is_unauthorized(self):
    '''
    Standard users cannot access to the API
    '''
    pass
def test_admin_user_can_see_the_api(self):
    '''
    Admin user can access to the API
    '''
    pass
```

Reply:
Hmm. We are using `subtest` here. Which handles this parametrization for us for each case. So, for debugging when a test fails, test specify the case where it failed. This also helps readability also. We read far less code.
https://docs.python.org/3/library/unittest.html#subtests